scripts/SREP/cluster-operator-status/script.sh

```diff
@@ -0,0 +1,114 @@
+#!/bin/bash
+# This script provides a rich summary of information concerning the status of 
+# degraded/unavailable OpenShift cluster operators.
+#
+# The script can specifically target a cluster operator through
+# the use of a "CLUSTER_OPERATOR" parameter
+
+set -e
+set -o nounset
+set -o pipefail
+
+
+# print_header prints a formatted section break header
+print_header() {
+    clusteroperator=$1
+    heading=$2
+    echo -e "--- [ CLUSTER OPERATOR '${clusteroperator}' ${heading} ] ---\n"
+}
+
+# get_summary prints the basic status for the given cluster operator parameter
+get_summary() {
+    clusteroperator=$1
+
+    # shellcheck disable=SC2016
+    available=$(jq -r --arg CLUSTER_OPERATOR "${clusteroperator}" '.items[] | select(.metadata.name == $CLUSTER_OPERATOR) | .status.conditions[] | select(.type == "Available") | .status' <<< "$OPERATOR_REPORT")
+    # shellcheck disable=SC2016
+    progressing=$(jq -r --arg CLUSTER_OPERATOR "${clusteroperator}" '.items[] | select(.metadata.name == $CLUSTER_OPERATOR) | .status.conditions[] | select(.type == "Progressing") | .status' <<< "$OPERATOR_REPORT")
+    # shellcheck disable=SC2016
+    degraded=$(jq -r --arg CLUSTER_OPERATOR "${clusteroperator}" '.items[] | select(.metadata.name == $CLUSTER_OPERATOR) | .status.conditions[] | select(.type == "Degraded") | .status' <<< "$OPERATOR_REPORT")
+
+    print_header "${clusteroperator}" "SUMMARY"
+    echo " - AVAILABLE:   ${available}"
+    echo " - DEGRADED:    ${degraded}"
+    echo " - PROGRESSING: ${progressing}"
+    echo -e "\n"
+}
+
+# get_description prints a detailed status description for the given cluster 
+# operator parameter
+get_description() {
+    clusteroperator=$1
+
+    if ! co_desc_report=$(oc describe clusteroperator "${clusteroperator}"); then
+       echo "An error was returned when describing ClusterOperator '${clusteroperator}': ${co_desc_report}"
+       return
+    fi
+    print_header "${clusteroperator}" "DESCRIPTION"
+    echo -e "${co_desc_report}\n\n"
+}
+
+# get_logs prints the last 15 lines of logs from the cluster operator pod
+get_logs() {
```

Comment:
Right now, the logs are mostly collected for the operator pod. In case of Monitoring stack and the machine-api CO, several pods are present and each can have multiple containers in it. What are your thoughts on showing logs for all containers too? Else, another option can be to have it for the detailed SOP for the respective CO as well.

Reply:
the standard of the container name that usually does the operations is that it's name is the same name as the deployment

this code can also be tested on the monitoring CO and see if it fails with the current implementation

Reply:
I've added `--all-containers` to the `oc logs` call now.